tools/osbuilder/initrd-builder/initrd_builder.sh

```diff
@@ -74,7 +74,13 @@ OK "Agent is installed"
 
 # initramfs expects /init, create symlink only if ${ROOTFS}/init does not exist
 # Init may be provided by other packages, e.g. systemd or GPU initrd/rootfs
-[ -x "${ROOTFS}/init" ] || [ -L "${ROOTFS}/init" ] || sudo ln -sf /sbin/init "${ROOTFS}/init"
+# ATTN: initrd /init cannot follow two levels of symlinks (/init to /sbin/init to /lib/systemd/systemd)
+# Setting /init directly to /lib/systemd/systemd when AGENT_INIT is disabled
+if [ ! -x "${ROOTFS}/init" ] && [ ! -L "${ROOTFS}/init" ]; then
+  [ "${AGENT_INIT}" = "yes" ] && \
```

Comment:
Hi @ryansavino !

`A && B || C` isn't same as if-then-else sometimes. Suppose that `A` is true, `B` run and fail, then `C` is executed.

I learned it long ago, you are still young so you still can learn too :)

Reply:
Addressed. Thanks.